libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2555,7 +2555,7 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 
 	if (possibleMembers.empty())
 	{
-		if (initialMemberCount == 0)
+		if (initialMemberCount == 0 && !dynamic_cast<ArraySliceType const*>(exprType))
```

Comment:
This solution is very specific to the issue that this fixes. I am open to suggestions to make this more generic to work with all types that would fail for a `withLocation()` call

Reply:
It's not pretty, but this code will be gone with 0.7.0, so I would say it's fine.